lib/src/functions/math.dart

```diff
@@ -289,11 +289,23 @@ final _random = math.Random();
 
 final _randomFunction = _function("random", r"$limit: null", (arguments) {
   if (arguments[0] == sassNull) return SassNumber(_random.nextDouble());
-  var limit = arguments[0].assertNumber("limit").assertInt("limit");
-  if (limit < 1) {
+  var limit = arguments[0].assertNumber("limit");
+
+  if (limit.hasUnits) {
+    warn(
+      "math.random() will no longer ignore \$limit units ($limit) in a "
+      "future release.\n"
+      "\n"
+      "If you meant to preserve units: "
+      "math.random($limit) * 1${limit.unitString}",
```

Comment:
It's generally a good idea to use parameter names rather than values when providing suggestions for built-in functions, since users are probably not just passing in literal values to begin with. (So `\$limit` rather than `$limit`.)

Reply:
yeah I was thinking something similar when looking at the tests 😅